shark_turbine/kernel/ops/wave_ops.py

```diff
@@ -198,6 +206,25 @@ def copy(self, new_name: Optional[str] = None) -> Self:
             new_node.name = new_name
         return get_custom(new_node)
 
+    def copy_to_new_graph(
+        self, new_graph: fx.Graph, new_name: Optional[str] = None
+    ) -> Self:
+        """Returns a duplicate of this node."""
+        new_node = new_graph.node_copy(self.fx_node)
+        new_node.tkw_op = self
+        if new_name:
+            new_node.name = new_name
+        return get_custom(new_node)
+
+    def replace_all_uses_with(self, new_node: CustomOp):
+        """Replace all uses of the current node with the new node."""
+        for user in self.users:
+            user.update_arg(user.node_args.index(self), new_node)
```

Comment:
The `update_arg` works both with instances of `CustomOp` and `fx.Node` as new values. Hence, we could change the signature to `def replace_all_uses_with(self, new_node: CustomOp | fx.Node):` and explicitly support that as well. In fact, we use this below already in `custom_node.replace_all_uses_with(promoted_read)` where `promoted_read` is an instance of `fx.Node`

Reply:
Cool, that's a nice side effect :) Will update the signature.